src/main/java/uk/gov/hmcts/darts/authentication/component/SoapRequestInterceptor.java

```diff
@@ -249,11 +250,14 @@ public void afterCompletion(MessageContext messageContext, Object endpoint, Exce
     }
 
     public void logCookieInformation() {
-        HttpServletRequest curRequest =
-            ((ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()).getRequest();
-
-        if (log.isTraceEnabled()) {
-            getCookieInformation(RESPONSE_PAYLOAD_PREFIX + "{}", curRequest);
+        try {
+            HttpServletRequest curRequest =
+                ((ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()).getRequest();
+            if (log.isTraceEnabled()) {
+                getCookieInformation(RESPONSE_PAYLOAD_PREFIX + "{}", curRequest);
+            }
+        } catch (Exception e) {
```

Comment:
I have tested a malformed cookie and works fine by catching a IllegalArgumentException. I guess you will need a PMD/sonar qube exclusion to catch Exception

Reply:
i'm thinking its a belt and braces approach, but was a bit of a gamble as to whether sonar would pass it, if it fails, i'll just remove it